### PR TITLE
alfalfa-import utility created.

### PR DESCRIPTION
**Pull request by sadjad**

### CODE REVIEW

File: src/decoder/frame_info.cc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+{
+  frame_.clear();
+  output_raster_.clear();
+}
+
+std::string build_frame_name( const SourceHash & source_hash,
+  const TargetHash & target_hash )
+{
+  stringstream ss;
+  ss << source_hash.str() << "#" << target_hash.str();
```
Comment (shacklettbp) [inline src/decoder/frame_info.cc]: source_hash.str() + "#" + target_hash.str()

File: src/decoder/frame_info.hh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  std::vector<uint8_t> frame_;
+  Optional<RasterHandle> output_raster_;
+
+public:
+  std::string frame_name;
+  Optional<std::string> ivf_filename;
+  size_t offset;
+  size_t length;
+  SourceHash source_hash;
+  TargetHash target_hash;
```
Comment (shacklettbp) [inline src/decoder/frame_info.hh]: This breaks the convention throughout the code of having all class member variables end with _. Can you just change these to getters and make them private?

File: src/frontend/Makefile.am [hunk 3]
```diff
@@ -23,3 +23,6 @@ frameplayer_LDADD = ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.
 
 collisions_SOURCES = collisions.cc
 collisions_LDADD = ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.a ../display/libalfalfadisplay.a ../util/libalfalfautil.a $(X264_LIBS)
+
+alfalfa_import_SOURCES = alfalfa-import.cc
+alfalfa_import_LDADD = ../format/libalfalfavideo.a ../format/protobufs/libalfalfaprotobufs.a ../decoder/libalfalfadecoder.a ../encoder/libalfalfaencoder.a ../util/libalfalfautil.a $(X264_LIBS) $(PROTOBUF_LIBS) -lboost_system -lboost_filesystem
```
Comment (shacklettbp) [inline src/frontend/Makefile.am]: :(

Comment (sadjad) [inline src/frontend/Makefile.am]: Got rid of boost-filesystem ;)

**Comment by sadjad** [comment]
There are some necessary changes that Deepak and I going to make, and I'm going to close this pull request. I will make a new pull request as soon as they're done (hopefully very soon!).

File: src/decoder/frame_info.cc (hunk 1 above)
Comment (sadjad) [inline src/decoder/frame_info.cc]: Done.

File: src/decoder/frame_info.hh (hunk 2 above)
Comment (sadjad) [inline src/decoder/frame_info.hh]: Done.

**Comment by keithw** [comment]
Ok! Also: let's not commit auto-generated protobuf files into the repository. Instead, they should be built as part of the build system. (See https://github.com/ravinet/mahimahi/tree/master/src/protobufs for how to do this.)

Otherwise we have a problem where our code will stop compiling once the user upgrades their version of protobuf...

**Comment by sadjad** [comment]
They're not, they're in the .gitignore! I think I used the exact same approach as Mahimahi. https://github.com/sadjad/alfalfa/blob/e035117d25e83c1409532985b16f2a9dcf27d67e/.gitignore

**Comment by keithw** [comment]
Whoops, apparently I read too fast!

**Comment by sadjad** [comment]
We changed the way we were storing databases to make them ready to create `alfalfa-combine` and other necessary commands. We also squashed all the commits together.

**Pull request merged**